caffe2/utils/Metaprogramming.h

```diff
@@ -23,9 +23,44 @@ struct function_traits<Result (Args...)> {
   using func_type = Result (Args...);
   using return_type = Result;
   using parameter_types = typelist::typelist<Args...>;
+  static constexpr auto number_of_parameters = sizeof...(Args);
```

Comment:
nit: there's a `typelist::size` that can be used on `parameter_types`. Not sure if replicating it here makes sense but I also don't mind if you want to keep it.

Reply:
I found it more cumbersome to write `c10::guts::typelist::size<typename Traits::parameter_types>` instead of `Traits::number_of_parameters`